packages/consumption/src/modules/requests/utility/validateRelationshipAttributesWithinRequest.ts

```diff
@@ -0,0 +1,206 @@
+
+    for (const item of requestItemGroup.items) {
+        if (item instanceof RequestItemGroup) {
+            const fragments = extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(item, recipient);
+            if (fragments) fragmentsOfMustBeAcceptedItemsOfGroup.push(...fragments);
+        } else {
+            const fragment = extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(item, recipient);
+            if (fragment) fragmentsOfMustBeAcceptedItemsOfGroup.push(fragment);
+        }
+    }
+
+    if (fragmentsOfMustBeAcceptedItemsOfGroup.length !== 0) return fragmentsOfMustBeAcceptedItemsOfGroup;
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(requestItem: RequestItem, recipient?: CoreAddress): RelationshipAttributeFragment | void {
+    if (requestItem.mustBeAccepted) {
+        return extractRelationshipAttributeFragmentFromRequestItem(requestItem, recipient);
+    }
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentsFromAcceptedItemsOfGroup(
+    requestItemGroup: RequestItemGroup,
+    decideGroupParams: DecideRequestItemGroupParametersJSON
+): RelationshipAttributeFragment[] | void {
+    const fragmentsOfAcceptedItemsOfGroup: RelationshipAttributeFragment[] = [];
+
+    for (let i = 0; i < decideGroupParams.items.length; i++) {
+        const decideItemParams = decideGroupParams.items[i];
+        const item = requestItemGroup.items[i];
+
+        if (item instanceof RequestItem) {
+            const fragment = extractRelationshipAttributeFragmentFromAcceptedRequestItem(item, decideItemParams);
+            if (fragment) fragmentsOfAcceptedItemsOfGroup.push(fragment);
+        }
+    }
+
+    if (fragmentsOfAcceptedItemsOfGroup.length !== 0) return fragmentsOfAcceptedItemsOfGroup;
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentFromAcceptedRequestItem(
+    requestItem: RequestItem,
+    decideItemParams: DecideRequestItemParametersJSON
+): RelationshipAttributeFragment | void {
+    if (decideItemParams.accept) {
+        return extractRelationshipAttributeFragmentFromRequestItem(requestItem);
+    }
+
+    return;
+}
+
+function extractRelationshipAttributeFragmentFromRequestItem(requestItem: RequestItem, recipient?: CoreAddress): RelationshipAttributeFragment | void {
+    if (requestItem instanceof CreateAttributeRequestItem && requestItem.attribute instanceof RelationshipAttribute) {
+        const ownerIsEmptyString = requestItem.attribute.owner.toString() === "";
+        return {
+            owner: ownerIsEmptyString && recipient ? recipient.toString() : requestItem.attribute.owner.toString(),
+            key: requestItem.attribute.key,
+            value: { "@type": requestItem.attribute.value.toJSON()["@type"] }
+        };
+    } else if ((requestItem instanceof ReadAttributeRequestItem || requestItem instanceof ProposeAttributeRequestItem) && requestItem.query instanceof RelationshipAttributeQuery) {
```

Comment:
I think the `else` is not required, since there is a `return` in the former `if` block.